java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -300,6 +307,20 @@ public static long writeBatchBuffers(WriteChannel out, ArrowRecordBatch batch) t
     return out.getCurrentPosition() - bufferStart;
   }
 
+  /**
+   * Serialize the compression body.
+   */
+  public static long writeCompressionBody(
+      WriteChannel out, ArrowBodyCompression bodyCompression) throws IOException {
+    long bufferStart = out.getCurrentPosition();
+    ByteBuffer buf = ByteBuffer.allocate(2);
```

Comment:
I wonder if rather than allocating a new ByteBuffer it might be easier to just use a byte[] and the `write(byte[])` method. Potentially you could then use a byte[] to store the data in `ArrowBodyCompression` so that you don't have to allocate anything while writing the message.

Reply:
I have revised the implementation of `ArrowBodyCompression ` so that it is based on byte[]. Thanks a lot for the good suggestion.